src/reanimated2/component/SkipInitialEnteringAnimations.tsx

```diff
@@ -0,0 +1,21 @@
+import { createContext, useEffect, useRef } from 'react';
+
+export const SkipEnteringContext =
+  createContext<React.MutableRefObject<boolean> | null>(null);
+
+export function SkipInitialEnteringAnimations(props: {
+  value?: boolean;
+  children?: React.ReactNode;
+}) {
+  const skipValueRef = useRef(props.value !== false);
```

Comment:
it doesn't work when `value` is `null` or `undefined`
<img width="184" alt="image" src="https://github.com/software-mansion/react-native-reanimated/assets/36106620/bbfffdb2-efd0-4e7c-b484-e5b6456a5fd3">

I think we should revers check to `props.value === true`

Reply:
I want the ref to be true by default (i.e. when props.value === undefined), that's why I went with that. This way you can just write `<Skip...>` with no props and it will remove the animation.

Reply:
Ok, I see. Let's add comment that this is a intentional behaviour